event_routing_backends/helpers.py

```diff
@@ -0,0 +1,140 @@
+"""
+Helper utilities for event transformers.
+"""
+import logging
+from datetime import timedelta
+from urllib.parse import parse_qs, urlparse
+
+from dateutil.parser import parse
+from django.contrib.auth import get_user_model
+from isodate import duration_isoformat
+
+# Imported from edx-platform
+# pylint: disable=import-error
+from lms.djangoapps.courseware.courses import get_course_by_id
+from opaque_keys.edx.keys import CourseKey
+from student.models import anonymous_id_for_user
+
+logger = logging.getLogger(__name__)
+User = get_user_model()
+
+UTC_DATETIME_FORMAT = '%Y-%m-%dT%H:%M:%S.%f'
+
+
+def get_anonymous_user_id_by_username(username, course_id=None):
+    """
+    Generate anonymous user id.
+
+    Generate anonymous id as per course if `course_id` is provided. Otherwise
+    generate anonymous id as per student.
+    If no anonymous id is generated, return "anonymous".
+
+    Arguments:
+        username (str):     username for the learner
+        course_id (str):    course key string.
+
+    Returns:
+        str
+    """
+    try:
+        user = User.objects.get(username=username)
+    except User.DoesNotExist:
+        logger.info('User with username "%s" does not exist. '
+                    'Cannot generate anonymous ID', username)
+
+        anonymous_id = 'anonymous'
+    else:
+        anonymous_id = anonymous_id_for_user(user, course_id)
+
+    return anonymous_id
+
+
+def get_course_from_id(course_id):
+    """
+    Get Course object using the `course_id`.
+
+    Arguments:
+        course_id (str) :   ID of the course
+
+    Returns:
+        Course
+    """
+    course_key = CourseKey.from_string(course_id)
+    return get_course_by_id(course_key)
+
+
+def convert_seconds_to_iso(seconds):
```

Comment:
I don't think this indirection for a 1 line function is necessary.

I disagree with this but not enough to block merging the change.